### PR TITLE
srtm elevation & tri computation scripts

### PR DESCRIPTION
**Pull request by Ohm-Np**
Hi Darius, 

Currently the scripts function like this:-

**`srtmelevation`**
- argument **stats** by default is _mean_, or else users could provide either _median_ or _sd_
- Do we need to provide all three stats at once for elevation or is it OK?
- added todisk
- default engine is `zonal`, or user could specify `extract`

**`tri`**
- argument **stats** by default is list of _mean_ ,  _median_ or _sd_
- Do we need to provide only one if users are interesed in only one stats?
- added todisk
- default engine is `zonal`, or user could specify `extract`

### CODE REVIEW

File: R/calc_TRI.R [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    overwrite = TRUE
+  )
+  tri <- terra::terrain(rast_mask,
+    v = "TRI",
+    unit = "degrees",
+    neighbors = 8,
+    filename = ifelse(todisk, file.path(rundir, "terrain.tif"), ""),
+    overwrite = TRUE
+  )
+  tri.mean <- terra::zonal(tri,
```
Comment (goergen95) [inline R/calc_TRI.R:65]: As noted in your comment it is best to include some if-statements to calculate mean/median/sd only if specified by the user. This also means that the stats specified should be checked for correctness at the top of the function

File: R/calc_TRI.R [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    overwrite = TRUE
+  )
+  tri <- terra::terrain(rast_mask,
+    v = "TRI",
+    unit = "degrees",
+    neighbors = 8,
+    filename = ifelse(todisk, file.path(rundir, "terrain.tif"), ""),
+    overwrite = TRUE
+  )
+  tri.mean <- terra::extract(tri,
```
Comment (goergen95) [inline R/calc_TRI.R:111]: same as the comment above applies here

File: R/calc_srtmElevation.R [hunk 3]
```diff
@@ -0,0 +1,86 @@
+#' Calculate elevation based on SRTM rasters
+#'
+
+.calc_SRTMElevation <- function(shp,
```
Comment (goergen95) [inline R/calc_srtmElevation.R:4]: I would suggest to rename the indicator to `.calc_dem` if that is ok with you

File: R/calc_srtmElevation.R [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#'
+
+.comp_eleExtract <- function(elevation = NULL,
+                             shp = NULL,
+                             stats = "mean",
+                             ...) {
+  shp.v <- vect(shp)
+  zstats <- terra::extract(elevation,
+    shp.v,
+    fun = stats,
```
Comment (goergen95) [inline R/calc_srtmElevation.R:45]: I suggest to also allow users to compute mean/median/sd for the DEM. Similar to the comments in the TRI routine, this should include a check that the stats were correctly specified and only calculate the one that a user is interested in.

File: R/calc_srtmElevation.R [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  )
+  p_raster <- terra::rasterize(shp.v,
+    rast_mask,
+    field = 1:nrow(shp.v),
+    filename =  ifelse(todisk, file.path(rundir, "polygon.tif"), ""),
+    overwrite = TRUE
+  )
+  zstats <- terra::zonal(rast_mask,
+    p_raster,
+    fun = stats,
```
Comment (goergen95) [inline R/calc_srtmElevation.R:78]: see the comment above.

File: R/indicator_backlog.R [hunk 6]
```diff
@@ -1,28 +1,47 @@
-available_indicators <- function(indicator = NULL){
```
Comment (goergen95) [inline R/indicator_backlog.R:1]: we got a merge issue in this function that should be resolved before merging to main

Comment (Ohm-Np) [reply R/indicator_backlog.R:1]: thank you for the super-fast response. I will soon update the changes

File: R/calc_TRI.R (hunk 1 above)
Comment (Ohm-Np) [reply R/calc_TRI.R:65]: Should we also give a try for `exact` engine using `exactextractr`?

**Review by goergen95 — CHANGES_REQUESTED** [review]
Please see my comments for the functions. Specifically, before merging the following issues need to be addressed:
- all user specified stats should be checked for correctness
- only the user specified stats should be calculated
- please consider renaming `.calc_srtmElevation` to `.calc_dem`. 
- also, I implemented a pre-commit hook (we can discuss this in person) to adhere to the tidyverse style guide. That means for variables and function names we should stick to the snake case style, that is only small letters and words should be seperated by underscores `_`, e.g. `.comp_tri_zonal()` or `shp_v`
- there are some merge conflicts in the indicator backlog that we need to address before merging

File: R/calc_TRI.R (hunk 1 above)
Comment (goergen95) [reply R/calc_TRI.R:65]: sure, another engine sounds great!

File: R/indicator_backlog.R [hunk 7]
```diff
@@ -1,28 +1,47 @@
...
+    ),
+    elevation = list(
+      name = ".calc_srtmElevation",
+      inputs = list(srtmelevation = "raster"),
+      arguments = list(
+        stats = "mean"
+      )
+    ),
+    tri <- list(
+      name = ".cal_TRI",
```
Comment (goergen95) [inline R/indicator_backlog.R:35]: catched a small typo here. guess its supposed to be `.calc_tri` ;)

File: R/get_climaticVariables.R [hunk 8]
```diff
@@ -1,8 +1,26 @@
+#' WorldClim climatic variables (min temperature, max temperature, mean precipitation)
+#'
+#' This resource is published by Fick et al. (2017) "WorldClim 2: new 1-km
+#' spatial resolution climate surfaces for global land areas". This resource
+#' represents multiple climatic variables from which we will be requiring minimum
+#' temperature, maximum temperature, and mean precipitation layers. The layers are
+#' available to download for the period 2000 - 2018 on monthly basis from WorldClim.
+#' Encoded as (°C), representing the minimum and maximum temperature per output grid
+#' cell while encoded as (mm), representing the mean precipitation per output grid cell.
+#'
```
Comment (goergen95) [inline R/get_climaticVariables.R:10]: This documents all three resources. Could you please add a itemized list that the users learn about that they can request three different resources

**Review by goergen95 — APPROVED** [review]

**Pull request merged**